### PR TITLE
feat: Support `SProd` and `CompositeOp` for expval

### PR DESCRIPTION
**Pull request by speller26**
*Issue #, if available:*
#253 
*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/amazon-braket/amazon-braket-pennylane-plugin-python/blob/main/README.md) and [API docs](https://github.com/amazon-braket/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by speller26** [comment]
@astralcai Feel free to check if this implementation makes sense.

**Comment by astralcai** [comment]
Since it's passing PL integration tests I say it's good.

File: src/braket/pennylane_plugin/translation.py [hunk 1]
```diff
@@ -548,7 +549,9 @@ def translate_result_type(  # noqa: C901
         then this will return a result type for each term.
     """
     return_type = measurement.return_type
+    targets = targets or measurement.wires.tolist()
     observable = measurement.obs
+    print(observable)
```
Comment (math411) [inline src/braket/pennylane_plugin/translation.py:554]:
```suggestion
```

File: src/braket/pennylane_plugin/translation.py [hunk 2]
```diff
@@ -560,93 +563,88 @@ def translate_result_type(  # noqa: C901
...
+        return Sample(braket_observable)
     else:
         raise NotImplementedError(f"Unsupported return type: {return_type}")
 
 
+def _flatten_observable(observable):
+    if isinstance(observable, (qml.ops.Hamiltonian, qml.ops.CompositeOp, qml.ops.SProd)):
+        simplified = qml.ops.LinearCombination(*observable.terms()).simplify()
+        coeffs, _ = simplified.terms()
+        if len(coeffs) > 1 or coeffs[0] != 1:
```
Comment (math411) [inline src/braket/pennylane_plugin/translation.py:593]: Why do we skip this if `coeffs == 1`?

Comment (speller26) [reply src/braket/pennylane_plugin/translation.py:592]: In this case, the observable is a flat tensor product, which we support directly.

**Review by math411 — APPROVED** [review]

**Pull request merged**